gadgets/trace_dns/test/integration/trace_dns_test.go

```diff
@@ -72,25 +73,25 @@ func TestTraceDNS(t *testing.T) {
 	serverImage := "ghcr.io/inspektor-gadget/dnstester:latest"
 	clientImage := "docker.io/library/busybox:latest"
 
-	// TODO: The current logic creates the namespace when running the pod, hence
-	// we need a namespace for each pod
-	var nsClient, nsServer string
+	var nsTest string
 	serverContainerOpts := []containers.ContainerOption{containers.WithContainerImage(serverImage)}
 	clientContainerOpts := []containers.ContainerOption{containers.WithContainerImage(clientImage)}
 
 	if utils.CurrentTestComponent == utils.KubectlGadgetTestComponent {
-		nsClient = utils.GenerateTestNamespaceName(t, "test-trace-dns-client")
-		clientContainerOpts = append(clientContainerOpts, containers.WithContainerNamespace(nsClient))
-
-		nsServer = utils.GenerateTestNamespaceName(t, "test-trace-dns-server")
-		serverContainerOpts = append(serverContainerOpts, containers.WithContainerNamespace(nsServer))
+		nsTest = utils.GenerateTestNamespaceName(t, "test-trace-dns")
+		clientContainerOpts = append(clientContainerOpts, containers.WithContainerNamespace(nsTest))
+		clientContainerOpts = append(clientContainerOpts, containers.WithUseExistingNamespace())
+		serverContainerOpts = append(serverContainerOpts, containers.WithContainerNamespace(nsTest))
+		serverContainerOpts = append(serverContainerOpts, containers.WithUseExistingNamespace())
+		testutils.ExecuteCreateTestNamespaceCommand(t, nsTest)
```

Comment:
- It could be named `CreateK8sNamespace()`
- Internally it could register the deletion of the namespace using `t.Cleanup()`, so the caller doesn't have to worry about it.
- It should be executed right after `nsTest = utils.GenerateTestNamespaceName(t, "test-trace-dns")`